firebase_admin/mlkit.py

```diff
@@ -18,8 +18,91 @@
 deleting, publishing and unpublishing Firebase ML Kit models.
 """
 
+import re
+import requests
+import six
+
+import firebase_admin
+from firebase_admin import _http_client
+from firebase_admin import _utils
+from firebase_admin import exceptions
+
+_MLKIT_ATTRIBUTE = '_mlkit'
+
+def _get_mlkit_service(app):
+    """ Returns an _MLKitService instance for an App.
+
+    Args:
+      app: A Firebase App instance (or None to use the default App).
+
+    Returns:
+      _MLKitService: An _MLKitService for the specified App instance.
+
+    Raises:
+      ValueError: If the app argument is invalid.
+    """
+    return _utils.get_app_service(app, _MLKIT_ATTRIBUTE, _MLKitService)
+
+def get_model(model_id, app=None):
+    mlkit_service = _get_mlkit_service(app)
+    return Model(mlkit_service.get_model(model_id))
+
+class Model(object):
+    """A Firebase ML Kit Model object."""
+    def __init__(self, data):
+        """Created from a data dictionary."""
+        self._data = data
+
+    def __eq__(self, other):
+        if isinstance(other, self.__class__):
+            return self._data == other._data
+        else:
+            return False
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    #TODO(ifielker): define the Model properties etc
+
 class _MLKitService(object):
     """Firebase MLKit service."""
 
-    BASE_URL = 'https://mlkit.googleapis.com'
-    PROJECT_URL = 'https://mlkit.googleapis.com/projects/{0}/'
+    BASE_URL = 'https://mlkit.googleapis.com/v1beta1/'
+    PROJECT_URL = 'https://mlkit.googleapis.com/v1beta1/projects/{0}/'
+
+    def __init__(self, app):
+        project_id = app.project_id
+        if not project_id:
+            raise ValueError(
+                'Project ID is required to access MLKit service. Either set the '
+                'projectId option, or use service account credentials.')
+        self._project_url = _MLKitService.PROJECT_URL.format(project_id)
+        self._client = _http_client.JsonHttpClient(credential=app.credential.get_credential())
+
+    def _request(self, method, urlpath, **kwargs):
+        """Makes an HTTP call using the Python requests library.
+
+        Args:
+            method: HTTP method name as a string (e.g. get, post, patch, delete).
+            urlpath: URL path to the endpoint. This will be appended to the
+                server's base project URL.
+            kwargs: An additional set of keyword arguments to be passed into requests
+                API (e.g. json, params)
+
+        Returns:
+          dict: The parsed JSON response.
+        """
+        return self._client.body(method, url=self._project_url + urlpath, **kwargs)
+
+    def get_model(self, model_id):
+        if not model_id:
```

Comment:
This can be removed. If model_id is None, your type check will catch it. If it's empty your regex check will catch it.

Reply:
done